pydicom/tests/test_uid.py

```diff
@@ -112,25 +40,180 @@ def testGenerateUID(self):
             self.assertEqual(uid[:len(valid_prefix)], valid_prefix)
             self.assertTrue(len(uid) <= 64)
 
-    def testIsValid(self):
+
+class TestUID(unittest.TestCase):
```

Comment:
when I was referring to fixutres maybe I was using the wrong term. But I actually was referring to [this fixture-parametrize](https://docs.pytest.org/en/latest/parametrize.html).

I would change the tests and write something like this
```py
@pytest.mark.parametrize("uid", [UID('1.2.840.10008.1.2'),
                                 '1.2.840.10008.1.2',
                                 ' VR Little Endian',])
def test_valid_uid(uid):
       assert self.uid == uid


@pytest.mark.parametrize("uid", [UID('1.2.840.10008.1.2.1'),
                                                      '1.2.840.10008.1.2.1',
                                                      'Explicit VR Little Endian',])
def test_invalid_uid(uid):
        assert self.uid != uid 
        # or test for the raise or something..
```

regarding your question this is no longer needed
```python
if __name__ == "__main__":
    # ???
```